lib/Conversion/TorchToTcp/Elementwise.cpp

```diff
@@ -417,13 +417,207 @@ class ConvertAtenAtan2Op : public OpConversionPattern<AtenAtan2Op> {
   }
 };
 
+class ConvertAtenToDtypeOp : public OpConversionPattern<AtenToDtypeOp> {
+public:
+  using OpConversionPattern<AtenToDtypeOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenToDtypeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value input = op.getSelf();
+    auto inputType = input.getType().dyn_cast<torch::Torch::ValueTensorType>();
+    auto outputType = op.getType().dyn_cast<torch::Torch::ValueTensorType>();
+    RankedTensorType resultType =
+        OpConversionPattern<AtenToDtypeOp>::getTypeConverter()
+            ->convertType(op.getType())
+            .cast<RankedTensorType>();
+
+    if (!inputType || !outputType)
+      return rewriter.notifyMatchFailure(
+          op, "Expected Input/Output to be ValueTensorType");
+
+    auto elementType = inputType.getDtype();
+    if (!elementType.isIntOrFloat())
+      return rewriter.notifyMatchFailure(
+          op, "Input tensor must have integer or floating-point datatype");
+
+    // The non_blocking arg should be a constant `False`.
+    bool nonBlocking;
+    if (!matchPattern(op.getNonBlocking(), m_TorchConstantBool(&nonBlocking))) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: non_blocking arg must be a constant");
+    } else if (nonBlocking) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: non_blocking arg is expected to be false");
+    }
+
+    // The copy arg should be a constant `False`.
+    bool copy;
+    if (!matchPattern(op.getCopy(), m_TorchConstantBool(&copy))) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: copy arg must be a constant");
+    } else if (copy) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: copy arg is expected to be false");
+    }
+
+    // Only `none`, `contiguous` and `preserve` memory_format is supported.
+    if (!op.getMemoryFormat().getType().isa<Torch::NoneType>()) {
+      int64_t memoryFormat;
+      if (!matchPattern(op.getMemoryFormat(),
+                        m_TorchConstantInt(&memoryFormat)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: the memory format should be specified in "
+                "an integer constant");
+      if (memoryFormat != torch_upstream::MemoryFormat::Contiguous &&
+          memoryFormat != torch_upstream::MemoryFormat::Preserve)
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: only none, contiguous and preserve "
+                "memory_format is supported");
+    }
+
+    // FP -> FP
+    if (inputType.getDtype().isa<mlir::FloatType>() &&
+        outputType.getDtype().isa<mlir::FloatType>())
+      rewriter.replaceOpWithNewOp<tcp::CastOp>(
+          op, resultType, adaptor.getSelf(), IntegerAttr{}, IntegerAttr{});
+    // FP -> INT
+    else if (inputType.getDtype().isa<mlir::FloatType>()) {
+      if (outputType.getDtype().isSignlessInteger())
```

Comment:
Instead of checking for these conditions, you can directly use the method `getSignedness()` to get the signedness of the input and output types. That would simplify this and the code below a lot.